src/engraving/libmscore/score.cpp

```diff
@@ -5289,8 +5292,8 @@ MasterScore::MasterScore()
     metaTags().insert("creationDate", QDate::currentDate().toString(Qt::ISODate));
 }
 
-MasterScore::MasterScore(const MStyle& s)
-    : MasterScore{}
+MasterScore::MasterScore(const MStyle& s, std::shared_ptr<engraving::EngravingProject> project)
+    : MasterScore{project}
```

Comment:
so EngravingProject keeps a raw pointer to MasterScore, but MasterScore keeps a shared_ptr to EngravingProject? Feels a bit odd to me...but I'm not quite sure how an EngravingProject is supposed to behave.

Reply:
Step by step we will lead everything to how it should be :)